src/DebuggingFeature.ts

```diff
@@ -0,0 +1,80 @@
+import { BaseLanguageClient, StaticFeature, Neovim } from "coc.nvim";
+import {
+  ExecuteCommandParams,
+  ExecuteCommandRequest,
+  InitializeParams,
+  ServerCapabilities
+} from "vscode-languageserver-protocol";
+import { commands } from "coc.nvim";
+
+interface DebugAdapterStartResponse {
+  name: string;
+  uri: string;
+}
+
+export class DebuggingFeature implements StaticFeature {
+  private withVimspector = false;
+
+  constructor(private _nvim: Neovim, private _client: BaseLanguageClient) {}
+
+  async preInit(): Promise<void> {
+    this.withVimspector = (await this._nvim.getVar("loaded_vimpector")) === 1;
+  }
+
+  fillInitializeParams(params: InitializeParams): void {
+    if (this.withVimspector) {
+      if (params.capabilities.experimental == null) {
+        params.capabilities.experimental = {};
+      }
+      params.capabilities.experimental.debuggingProvider = true;
+    }
+  }
+
+  fillClientCapabilities(): void {}
+
+  public initialize(capabilities: ServerCapabilities): void {
+    if (
+      this.withVimspector &&
+      capabilities.experimental &&
+      capabilities.experimental.debuggingProvider
+    ) {
+      const debugHandler = async (debug: boolean, ...args: any[]) => {
+        let params: ExecuteCommandParams = {
+          command: "debug-adapter-start",
+          arguments: args
+        };
+        return this._client
+          .sendRequest(ExecuteCommandRequest.type, params)
+          .then(
+            async (resp: DebugAdapterStartResponse) => {
+              const colonIdx = resp.uri.lastIndexOf(":");
+              if (colonIdx !== -1) {
+                const port = parseInt(resp.uri.substr(colonIdx + 1));
+                if (!debug)
+                  await this._nvim.call("vimspector#ClearBreakpoints", []);
+                await this._nvim.call("vimspector#LaunchWithSettings", [
+                  { configuration: "cocmetals", test: "default", port }
```

Comment:
this looks wring

you're setting the `test` variable (which is apparently not used in the configuration) to 'default' (why?) and `port` appears to be an integer, where it should be `port: port`. how does this work ?

Reply:
Variable `test`. I forgot to remove after my tests. Removed.
Variable `port`. It is syntactic sugar which was added in ecmascript 6 which is equivalent to `port: port`.